mender-update/http_resumer.hpp

```diff
@@ -0,0 +1,179 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_UPDATE_HTTP_RESUMER_HPP
+#define MENDER_UPDATE_HTTP_RESUMER_HPP
+
+#include <string>
+#include <memory>
+#include <vector>
+
+#include <common/error.hpp>
+#include <common/io.hpp>
+#include <common/log.hpp>
+#include <common/events.hpp>
+#include <common/http.hpp>
+
+namespace mender {
+namespace update {
+namespace http_resumer {
+
+using namespace std;
+
+namespace error = mender::common::error;
+namespace io = mender::common::io;
+namespace log = mender::common::log;
+namespace events = mender::common::events;
+namespace http = mender::http;
+
+enum class DownloadResumerUserHandlersStatus {
+	None,
+	HeaderHandlerCalled,
+	BodyHandlerCalled,
+};
+
+struct DownloadResumerClientState {
+	bool initialized {false};
+	long long content_length {0};
+	long long offset {0};
```

Comment:
Why not `size_t`?

Reply:
`offset` could be `size_t`, but `content_length` I would still keep it signed (related discussion [here](https://github.com/mendersoftware/mender/pull/1364#discussion_r1309279151)) for easy parsing from the header. If we accept than, then is better that both are signed for comparison between then.

I really don't have an strong opinion. If you insist I will do it: I can very well parse content length from the header directly to `size_t` and propagate this also to the regular client's content length parsing.

Reply:
There's also `ssize_t`. :wink:

Reply:
Using `ssize_t` now.